File: widget/src/main/java/com/uploadcare/android/widget/data/Action.java
package com.uploadcare.android.widget.data;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Action {
    public static final String ACTION_OPEN_PATH ="open_path";
    public static final String ACTION_SELECT_FILE ="select_file";
    @SerializedName("obj_type")
    public String objectType;

    public String action;

    public String url;

    public Path path;

    @Override
    public String toString() {
        return "Action{" +
                "objectType='" + objectType + '\'' +
                ", action='" + action + '\'' +
                ", url='" + url + '\'' +
                ", path=" + path +
                '}';
    }

    public static class Path{
        public List<Chunk> chunks;

        @SerializedName("obj_type")
        public String objectType;

        @Override
        public String toString() {
            return "Path{" +
                    "chunks=" + chunks +
                    ", objectType='" + objectType + '\'' +
                    '}';
        }
    }

}

File: widget/src/main/java/com/uploadcare/android/widget/data/Action.kt
package com.uploadcare.android.widget.data

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

@Parcelize @Suppress("unused")
data class Action(@Json(name = "obj_type") val objectType: String,
                  val action: String?,
                  val url: String?,
                  val path: Path?) : Parcelable {
    companion object {
        const val ACTION_OPEN_PATH = "open_path"
        const val ACTION_SELECT_FILE = "select_file"
    }
}

@Parcelize
data class Path(@Json(name = "obj_type") val objectType: String,
                val chunks: List<Chunk>?) : Parcelable